<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to fastCNV - Visium HD data • fastCNV</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to fastCNV - Visium HD data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">fastCNV</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/fastCNV_HD.html">Introduction to fastCNV - Visium HD data</a></li>
    <li><a class="dropdown-item" href="../articles/fastCNV_sc.html">Introduction to fastCNV - scRNAseq data</a></li>
    <li><a class="dropdown-item" href="../articles/fastCNV_ST.html">Introduction to fastCNV - ST data</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to fastCNV - Visium HD data</h1>
            
      

      <div class="d-none name"><code>fastCNV_HD.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="context">Context<a class="anchor" aria-label="anchor" href="#context"></a>
</h2>
<p><code>fastCNV</code> is a package to detect the putative Copy Number
Variations (CNVs) in single cell (scRNAseq) data or Spatial
Transcriptomics (ST) data. It also plots the computed CNVs.</p>
<p>In this vignette you will learn how to run this package on Visium HD
seurat object. fastCNV runs on <strong>Seurat5</strong>.</p>
<p><strong>Warning</strong> : Visium HD samples are very RAM demanding
and the session may crash if not enough RAM is available. We recommend
running this vignette in a server.</p>
</div>
<div class="section level2">
<h2 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h2>
<p>To get the example data used in this vignette, you’ll need to install
and load the <code>fastCNVdata</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"must-bioinfo/fastCNV"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"must-bioinfo/fastCNVdata"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://must-bioinfo.github.io/fastCNV">fastCNV</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">fastCNVdata</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-the-data">Load the data<a class="anchor" aria-label="anchor" href="#load-the-data"></a>
</h2>
<p>Load your Visium HD data with Seurat.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HDBreast</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fastCNVdata/man/load_HDBreast.html" class="external-link">load_HDBreast</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="check-annotation">Check annotation<a class="anchor" aria-label="anchor" href="#check-annotation"></a>
</h2>
<p>You can load a separate annotation file for your dataset. It’s
usefull to have annotated spots for the analysis.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Import the annotation file corresponding to your sample. For 10X ST data you can annotate your spots with LoupeBrowser. The example data already has the annotations in it. </span></span>
<span><span class="va">annotation_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"/path/to/your/annotations/for/VisiumHD.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">HDBreast</span><span class="op">[[</span><span class="st">"annots_8um"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">annotation_file</span><span class="op">$</span><span class="va">Annotations</span></span></code></pre></div>
<p>You can take a look at which annotations are present in your objects,
in order to select the labels of the cells that will be used as
reference.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">HDBreast</span><span class="op">[[</span><span class="st">"annots_8um"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;                       annots_8um</span></span>
<span><span class="co">#&gt; s_016um_00107_00066-1           </span></span>
<span><span class="co">#&gt; s_008um_00269_00526-1    NoTumor</span></span>
<span><span class="co">#&gt; s_008um_00260_00253-1      Tumor</span></span></code></pre></div>
<p>fastCNV will use by default the 16µm assay, so if your annotations
are done on the 8µm resolution you will need to run this function before
running fastCNV on the 16µm assay. This function maps the annotations
from the 8µm resolution to the 16µm resolution.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HDBreast</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotations8umTo16um.html">annotations8umTo16um</a></span><span class="op">(</span><span class="va">HDBreast</span>, referenceVar <span class="op">=</span> <span class="st">"annots_8um"</span><span class="op">)</span> </span>
<span><span class="co">#&gt; [1] "New annotation column (new referenceVar) is named : projected_annots_8um."</span></span></code></pre></div>
<p>Now the 16um spots also have the annotations given to the 8um spots.
The projected annotations are in the meta data “projected_annots_8um”.
This will be the referenceVar we will use from now on.</p>
<p>You can check if the projected annotations are correct by doing</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialDimPlot</a></span><span class="op">(</span><span class="va">HDBreast</span>, group.by <span class="op">=</span> <span class="st">"projected_annots_8um"</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/spatialdimplot_annotations-1.png" alt="" width="768"></p>
</div>
<div class="section level2">
<h2 id="run-fastcnv">Run fastCNV<a class="anchor" aria-label="anchor" href="#run-fastcnv"></a>
</h2>
<p>During this step, fastCNV_10XHD() will use:</p>
<ul>
<li><p><code>CNVAnalysis</code> : computes the CNV. If you have
annotations for your Seurat object, you can add the parameters
<code>referenceVar</code> and <code>referenceLabel</code>.</p></li>
<li><p><code>CNVPerChromosome</code> : to compute the CNV per chromosome
arm, and store it in the metadata of the Seurat object. This part of
<code><a href="../reference/fastCNV.html">fastCNV()</a></code> can be skipped by turning
<code>getCNVPerChromosome</code> to <code>FALSE</code>.</p></li>
<li><p><code>CNVCluster</code> : OPTIONAL - performs hierarchical
clustering on a genomic score matrix extracted from a Seurat object.
This may crash with large Visium HD samples. This function can be run
later on.</p></li>
<li><p><code>plotCNVResultsHD</code> : to visualize the results (stored
in the assays slot of the Seurat object). This function will also build
a PNG or PDF file for each sample containing their corresponding CNV
heatmap in the current working directory, which can be changed using the
<code>savePath</code> parameter.</p></li>
</ul>
<p>We are going to run <code><a href="../reference/fastCNV.html">fastCNV()</a></code> on our Visium HD object.
The reference should be the non-tumor spots, here “NoTumor”.
<strong>Warning</strong> : This can be very RAM demanding.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HDBreast</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fastCNV_10XHD.html">fastCNV_10XHD</a></span><span class="op">(</span><span class="va">HDBreast</span>, sampleName <span class="op">=</span> <span class="st">"HDBreast"</span>, referenceVar <span class="op">=</span> <span class="st">"projected_annots_8um"</span>, referenceLabel <span class="op">=</span> <span class="st">"NoTumor"</span>, printPlot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/run1-1.png" alt="" width="768"></p>
</div>
<div class="section level2">
<h2 id="cnv-fraction">CNV fraction<a class="anchor" aria-label="anchor" href="#cnv-fraction"></a>
</h2>
<p>fastCNV also computes a <code>cnv_fraction</code> for each
observation in the Seurat object. This can be directly plotted using
Seurat spatial plotting functions, as we’ll demonstrate on STColon1:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">HDBreast</span>, <span class="st">"cnv_fraction"</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialPlot</a></span><span class="op">(</span><span class="va">HDBreast</span>, group.by <span class="op">=</span> <span class="st">"projected_annots_8um"</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/plot_cnv_fraction-1.png" alt="" width="768"></p>
<p>Here, we see some zones with higher CNV fractions than others. We can
directly plot and test this:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/FetchData.html" class="external-link">FetchData</a></span><span class="op">(</span><span class="va">HDBreast</span>, vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"projected_annots_8um"</span>, <span class="st">"cnv_fraction"</span><span class="op">)</span><span class="op">)</span>, </span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">projected_annots_8um</span>, <span class="va">cnv_fraction</span>, fill <span class="op">=</span> <span class="va">projected_annots_8um</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">1</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/plot_cnv_fraction2-1.png" alt="" width="576"></p>
<p>The tumor zone has a higher CNV fraction, which is coherent.</p>
<p>We can also plot the CNV per chromosome arm with the plotting
functions from Seurat.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org" class="external-link">scales</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">HDBreast</span>, features <span class="op">=</span> <span class="st">"11.q_CNV"</span><span class="op">)</span>  <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_distiller</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"RdBu"</span>, direction <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.5</span>, <span class="fl">0.5</span><span class="op">)</span>, </span>
<span>                       rescaler <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">to</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, <span class="va">from</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span></span>
<span>                         <span class="fu"><a href="https://scales.r-lib.org/reference/rescale_mid.html" class="external-link">rescale_mid</a></span><span class="op">(</span><span class="va">x</span>, to <span class="op">=</span> <span class="va">to</span>, mid <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span>                       <span class="op">}</span><span class="op">)</span> <span class="op">|</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">HDBreast</span>, features <span class="op">=</span> <span class="st">"8.q_CNV"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_distiller</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"RdBu"</span>, direction <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.5</span>, <span class="fl">0.5</span><span class="op">)</span>, </span>
<span>                       rescaler <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">to</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, <span class="va">from</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span></span>
<span>                         <span class="fu"><a href="https://scales.r-lib.org/reference/rescale_mid.html" class="external-link">rescale_mid</a></span><span class="op">(</span><span class="va">x</span>, to <span class="op">=</span> <span class="va">to</span>, mid <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span>                       <span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/plot_cnv_per_chromosome-1.png" alt="" width="672"></p>
</div>
<div class="section level2">
<h2 id="cnv-clusters">CNV clusters<a class="anchor" aria-label="anchor" href="#cnv-clusters"></a>
</h2>
<p>In addition to the CNV fraction, fastCNV also computes the CNV
clusters. These can be plotted with the Seurat plotting functions.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialDimPlot</a></span><span class="op">(</span><span class="va">HDBreast</span>, group.by <span class="op">=</span> <span class="st">"cnv_clusters"</span><span class="op">)</span> </span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/plot_cnv_clusters-1.png" alt="" width="480">
Here, we see many different CNV clusters with minimal variability. We
can compute the clusters only taking into account some annotations
(“tumor” here for example)</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To do the clustering with only certain annotations : </span></span>
<span><span class="va">HDBreast</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CNVCluster.html">CNVCluster</a></span><span class="op">(</span><span class="va">HDBreast</span>, referenceVar <span class="op">=</span> <span class="st">"projected_annots_8um"</span>,</span>
<span>                       cellTypesToCluster <span class="op">=</span> <span class="st">"Tumor"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialDimPlot</a></span><span class="op">(</span><span class="va">HDBreast</span>, group.by <span class="op">=</span> <span class="st">"cnv_clusters"</span><span class="op">)</span> </span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/new_clusters-1.png" alt="" width="480"></p>
<p>Or merge the clusters with great correlation.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To merge the clusters with high correlation</span></span>
<span><span class="va">HDBreast</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mergeCNVClusters.html">mergeCNVClusters</a></span><span class="op">(</span><span class="va">HDBreast</span>, mergeThreshold <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span> <span class="co">#adjust the thershold to your liking</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialDimPlot</a></span><span class="op">(</span><span class="va">HDBreast</span>, group.by <span class="op">=</span> <span class="st">"cnv_clusters"</span><span class="op">)</span> </span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/new_clusters2-1.png" alt="" width="480"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.github.io/seurat-object/" class="external-link">SeuratObject</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">HDBreast</span><span class="op">$</span><span class="va">cnv_clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">HDBreast</span><span class="op">$</span><span class="va">cnv_clusters</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/FetchData.html" class="external-link">FetchData</a></span><span class="op">(</span><span class="va">HDBreast</span>, vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cnv_clusters"</span>, <span class="st">"projected_annots_8um"</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">projected_annots_8um</span>, fill <span class="op">=</span> <span class="va">cnv_clusters</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">1</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/cnv_props-1.png" alt="" width="480"></p>
<p>Once we have the desired clusters, we can do the CNV heatmap again to
visualize the new clusters in it.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCNVResultsHD.html">plotCNVResultsHD</a></span><span class="op">(</span><span class="va">HDBreast</span>, referenceVar <span class="op">=</span> <span class="st">"projected_annots_8um"</span>, printPlot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/cnvheatmapt_clusters-1.png" alt="" width="768"></p>
</div>
<div class="section level2">
<h2 id="cnv-tree">CNV tree<a class="anchor" aria-label="anchor" href="#cnv-tree"></a>
</h2>
<p>In this part of the vignette, we will show how to plot the CNV
subclonality tree. CNV clusters are needed.</p>
<p>We build the tree, using <code><a href="../reference/CNVTree.html">CNVTree()</a></code>. It contains the
following functions :</p>
<ul>
<li><p><code>generateCNVClonesMatrix</code> will summarize
<code><a href="../reference/fastCNV.html">fastCNV()</a></code>’s results into a cluster/chromosome arm CNV
matrix.</p></li>
<li><p><code><a href="../reference/buildCNVTree.html">buildCNVTree()</a></code> will build the subclonality
tree.</p></li>
<li><p><code><a href="../reference/annotateCNVTree.html">annotateCNVTree()</a></code> will annotate each cnv gain/loss
in the tree.</p></li>
<li><p><code><a href="../reference/plotCNVTree.html">plotCNVTree()</a></code> will output the tree.</p></li>
</ul>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tree_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CNVTree.html">CNVTree</a></span><span class="op">(</span><span class="va">HDBreast</span>, values <span class="op">=</span><span class="st">"calls"</span>, cnv_thresh <span class="op">=</span> <span class="fl">0.09</span>, healthyClusters <span class="op">=</span> <span class="st">"1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_HD_files/figure-html/tree_cnv-1.png" alt="" width="672"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Gadea Cabrejas, Clarice Groeneveld.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
