<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to fastCNV - scRNAseq data • fastCNV</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to fastCNV - scRNAseq data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">fastCNV</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/fastCNV_HD.html">Introduction to fastCNV - Visium HD data</a></li>
    <li><a class="dropdown-item" href="../articles/fastCNV_sc.html">Introduction to fastCNV - scRNAseq data</a></li>
    <li><a class="dropdown-item" href="../articles/fastCNV_ST.html">Introduction to fastCNV - ST data</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to fastCNV - scRNAseq data</h1>
            
      

      <div class="d-none name"><code>fastCNV_sc.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="context">Context<a class="anchor" aria-label="anchor" href="#context"></a>
</h2>
<p><code>fastCNV</code> is a package to detect the putative Copy Number
Variations (CNVs) in single cell (scRNAseq) data or Spatial
Transcriptomics (ST) data. It also plots the computed CNVs.</p>
<p>In this vignette you will learn how to run this package on single
scRNAseq seurat objects and on a list of scRNAseq Seurat objects.
fastCNV runs on <strong>Seurat5</strong>.</p>
</div>
<div class="section level2">
<h2 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h2>
<p>To get the example data used in this vignette, you’ll need to install
and load the <code>fastCNVdata</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"must-bioinfo/fastCNV"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"must-bioinfo/fastCNVdata"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://must-bioinfo.github.io/fastCNV">fastCNV</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">fastCNVdata</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-the-example-data">Load the example data<a class="anchor" aria-label="anchor" href="#load-the-example-data"></a>
</h2>
<p><code>fastCNV</code> package works both on scRNAseq data or ST data,
and we will demonstrate on scRNAseq here, and there is another vignette
for ST data.</p>
<p>Here, we load scColon1, scColon2, scColon3, scColon4 (scRNAseq data
from colorectal tumors from Pelka et al. work).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">scColon1</span><span class="op">)</span></span>
<span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">scColon2</span><span class="op">)</span></span>
<span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">scColon3</span><span class="op">)</span></span>
<span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">scColon4</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="check-annotation">Check annotation<a class="anchor" aria-label="anchor" href="#check-annotation"></a>
</h2>
<p>You can load a separate annotation file for your dataset if this
information is not available in the Seurat object. For the data used in
this vignette, cell type annotations are already present under the
column <code>"annot"</code> of the metadata.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Import the annotation file corresponding to your sample. </span></span>
<span><span class="va">annotation_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"/path/to/your/annotations/for/scColon1.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">scColon1</span><span class="op">[[</span><span class="st">"annot"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">annotation_file</span><span class="op">$</span><span class="va">Annot</span></span></code></pre></div>
<p>You can take a look at which annotations are present in your objects,
in order to select the labels of the cells that will be used as
reference.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">scColon1</span><span class="op">[[</span><span class="st">"annot"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">scColon2</span><span class="op">[[</span><span class="st">"annot"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">scColon3</span><span class="op">[[</span><span class="st">"annot"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">scColon4</span><span class="op">[[</span><span class="st">"annot"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-fastcnv">Run fastCNV<a class="anchor" aria-label="anchor" href="#run-fastcnv"></a>
</h2>
<p>During this step, fastCNV() will use:</p>
<ul>
<li><p><code>prepareCountsForCNVAnalysis</code> : runs the Seurat
standard clustering algorithm and then aggregates the observations
(cells or spots) to into metaspots with up to the number of counts
defined by <code>aggregFactor</code> (default 15,000). In addition, the
observations can be aggregated on their seurat cluster AND their cell
type combined by leaving default <code>aggregateByVar = TRUE</code> and
specifying parameter <code>referenceVar</code>. If the Seurat object has
previously been clustered, the clustering will be re-done on
sctransformed (SCT) data using 10 PCs with default parameters to
<code>FindNeighbors</code> and <code>FindClusters</code>. This can be
skipped by setting <code>reClusterSeurat = FALSE</code>.</p></li>
<li><p><code>CNVAnalysis</code> : computes the CNV. If you have
annotations for your Seurat object, you can add the parameters
<code>referenceVar</code> and <code>referenceLabel</code>.</p></li>
<li><p><code>CNVPerChromosome</code> : to compute the CNV per chromosome
arm, and store it in the metadata of the Seurat object. This part of
<code><a href="../reference/fastCNV.html">fastCNV()</a></code> can be skipped by turning
<code>getCNVPerChromosome</code> to <code>FALSE</code>.</p></li>
<li><p><code>CNVCluster</code> : performs hierarchical clustering on a
genomic score matrix extracted from a Seurat object.</p></li>
<li><p><code>plotCNVResults</code> : to visualize the results (stored in
the assays slot of the Seurat object). By default, the parameter
<code>downsizePlot</code> is set to <code>FALSE</code>, which builds a
detailed plot, but takes an important time to render. If desired, you
can set the parameter to <code>TRUE</code>, which decreases the
rendering time by plotting the results at the meta-cell level instead of
the cell-level, thus decreasing the definition of the CNV results
plotted.<br>
This function will also build a PDF file for each sample containing
their corresponding CNV heatmap in the current working directory, which
can be changed using the <code>savePath</code> parameter.</p></li>
</ul>
<p>We are first going to run <code><a href="../reference/fastCNV.html">fastCNV()</a></code> on our scColon1
object, taking as reference the cells labeled as <code>TNKILC</code>,
<code>Myeloid</code>,<code>B</code>, <code>Mast</code> and
<code>Plasma</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scColon1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fastCNV.html">fastCNV</a></span><span class="op">(</span>seuratObj <span class="op">=</span> <span class="va">scColon1</span>, sampleName <span class="op">=</span> <span class="st">"scColon1"</span>, referenceVar <span class="op">=</span></span>
<span>                     <span class="st">"annot"</span>, referenceLabel <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TNKILC"</span>, <span class="st">"Myeloid"</span>, <span class="st">"B"</span>,</span>
<span>                     <span class="st">"Mast"</span>, <span class="st">"Plasma"</span><span class="op">)</span>, printPlot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_sc_files/figure-html/run1-1.png" alt="" width="720"></p>
<p>We are now going to run the <code><a href="../reference/fastCNV.html">fastCNV()</a></code> function on a list
of Seurat objects. When <code><a href="../reference/fastCNV.html">fastCNV()</a></code> is run on a list of
Seurat objects with the default parameters, it builds a pooled reference
between all the samples, which is particularly useful for data not
containing any healthy cells to build its own reference.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seuratList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">scColon2</span>,<span class="va">scColon3</span>,<span class="va">scColon4</span><span class="op">)</span></span>
<span><span class="va">sampleNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"scColon2"</span>, <span class="st">"scColon3"</span>, <span class="st">"scColon4"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">seuratList</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sampleNames</span></span>
<span><span class="va">referencelabels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Plasma"</span>, <span class="st">"TNKILC"</span>, <span class="st">"Myeloid"</span>, <span class="st">"B"</span>, <span class="st">"Mast"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seuratList</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fastCNV.html">fastCNV</a></span><span class="op">(</span><span class="va">seuratList</span>, <span class="va">sampleNames</span>, referenceVar <span class="op">=</span> <span class="st">"annot"</span>, </span>
<span>                      referenceLabel <span class="op">=</span> <span class="va">referencelabels</span>, printPlot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_sc_files/figure-html/run2-1.png" alt="" width="768"><img src="fastCNV_sc_files/figure-html/run2-2.png" alt="" width="768"><img src="fastCNV_sc_files/figure-html/run2-3.png" alt="" width="768"></p>
</div>
<div class="section level2">
<h2 id="cnv-fraction">CNV fraction<a class="anchor" aria-label="anchor" href="#cnv-fraction"></a>
</h2>
<p>fastCNV also computes a <code>cnv_fraction</code> for each
observation in the Seurat object. This can be directly plotted using
Seurat plotting functions, as we’ll demonstrate on scColon1:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="va">scColon1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">scColon1</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">common_theme</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>  plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>  legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>  legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>  axis.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>  axis.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">scColon1</span>, features <span class="op">=</span> <span class="st">"cnv_fraction"</span>, reduction <span class="op">=</span> <span class="st">"umap"</span> <span class="op">)</span> <span class="op">&amp;</span> <span class="va">common_theme</span> <span class="op">|</span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">scColon1</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span>  <span class="st">"annot"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">common_theme</span></span></code></pre></div>
<p><img src="fastCNV_sc_files/figure-html/plot_cnv_umap-1.png" alt="" width="768"></p>
<p>Here, we see some clusters with much higher CNV fractions than
others. We can directly plot and test this:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/FetchData.html" class="external-link">FetchData</a></span><span class="op">(</span><span class="va">scColon1</span>, vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"annot"</span>, <span class="st">"cnv_fraction"</span><span class="op">)</span><span class="op">)</span>, </span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">annot</span>, <span class="va">cnv_fraction</span>, fill <span class="op">=</span> <span class="va">annot</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">1</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_sc_files/figure-html/plot_cnv_fraction-1.png" alt="" width="672"></p>
<p>We can also plot the CNV per chromosome arm with the plotting
functions from Seurat.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org" class="external-link">scales</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">scColon1</span>, features <span class="op">=</span> <span class="st">"20.p_CNV"</span><span class="op">)</span>  <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_distiller</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"RdBu"</span>, direction <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                       rescaler <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">to</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, <span class="va">from</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span></span>
<span>                         <span class="fu"><a href="https://scales.r-lib.org/reference/rescale_mid.html" class="external-link">rescale_mid</a></span><span class="op">(</span><span class="va">x</span>, to <span class="op">=</span> <span class="va">to</span>, mid <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span>                       <span class="op">}</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">common_theme</span> <span class="op">|</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">scColon1</span>, features <span class="op">=</span> <span class="st">"X.q_CNV"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_distiller</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"RdBu"</span>, direction <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                       rescaler <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">to</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, <span class="va">from</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span></span>
<span>                         <span class="fu"><a href="https://scales.r-lib.org/reference/rescale_mid.html" class="external-link">rescale_mid</a></span><span class="op">(</span><span class="va">x</span>, to <span class="op">=</span> <span class="va">to</span>, mid <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span>                       <span class="op">}</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">common_theme</span></span></code></pre></div>
<p><img src="fastCNV_sc_files/figure-html/plot_cnv_per_chromosome-1.png" alt="" width="672"></p>
<p>Here, we observe that a subset of the epithelial tumor cells exhibit
a CNV loss in chromosome 20p, while others do not. Additionally, some
cells show a gain in chromosome Xp, whereas others experience a
loss.</p>
</div>
<div class="section level2">
<h2 id="cnv-classification">CNV classification<a class="anchor" aria-label="anchor" href="#cnv-classification"></a>
</h2>
<p>Using the CNV per chromosome arm and <code>CNVclassification()</code>
we can get the alterations per chromosome arm (gain, loss or no
alteration).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scColon1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CNVClassification.html">CNVClassification</a></span><span class="op">(</span><span class="va">scColon1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">scColon1</span>, group.by <span class="op">=</span> <span class="st">"20.p_CNV_classification"</span><span class="op">)</span> <span class="op">&amp;</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>gain <span class="op">=</span> <span class="st">"red"</span>, no_alteration <span class="op">=</span> <span class="st">"grey"</span>, loss <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span> <span class="op">&amp;</span></span>
<span>  <span class="va">common_theme</span> <span class="op">|</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">scColon1</span>, group.by <span class="op">=</span> <span class="st">"X.q_CNV_classification"</span><span class="op">)</span> <span class="op">&amp;</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>gain <span class="op">=</span> <span class="st">"red"</span>, no_alteration <span class="op">=</span> <span class="st">"grey"</span>, loss <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span> <span class="op">&amp;</span></span>
<span>  <span class="va">common_theme</span></span></code></pre></div>
<p><img src="fastCNV_sc_files/figure-html/cnv_classification-1.png" alt="" width="672"></p>
</div>
<div class="section level2">
<h2 id="cnv-clusters">CNV clusters<a class="anchor" aria-label="anchor" href="#cnv-clusters"></a>
</h2>
<p>In addition to the CNV fraction, fastCNV also computes the CNV
clusters and saves them in the Seurat object metadata as “cnv_clusters”.
These can be plotted with the Seurat plotting functions.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">scColon1</span>, group.by <span class="op">=</span> <span class="st">"cnv_clusters"</span><span class="op">)</span> <span class="op">+</span> <span class="va">common_theme</span></span></code></pre></div>
<p><img src="fastCNV_sc_files/figure-html/plot_cnv_clusters-1.png" alt="" width="384"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.github.io/seurat-object/" class="external-link">SeuratObject</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/FetchData.html" class="external-link">FetchData</a></span><span class="op">(</span><span class="va">scColon1</span>, vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cnv_clusters"</span>, <span class="st">"annot"</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">annot</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">cnv_clusters</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">1</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_sc_files/figure-html/cnv_props-1.png" alt="" width="576"></p>
<p>We find that tumor cells are virtually always cnv_cluster 2, 3, 4 or
5 ; meanwhile, healthy cells tend to be cnv_cluster 1 which present very
few or no CNV.</p>
</div>
<div class="section level2">
<h2 id="cnv-tree">CNV tree<a class="anchor" aria-label="anchor" href="#cnv-tree"></a>
</h2>
<p>As we can see, there are 4 tumor subclusters (2, 3, 4 and 5) in
scColon1, and 1 healthy subcluster (1).</p>
<p>In this part of the vignette, we will show how to plot the CNV
subclonality tree. CNV clusters are needed.</p>
<p>We build the tree, using <code><a href="../reference/CNVTree.html">CNVTree()</a></code>. It contains the
following functions :</p>
<ul>
<li><p><code>generateCNVClonesMatrix</code> will summarize
<code><a href="../reference/fastCNV.html">fastCNV()</a></code>’s results into a cluster/chromosome arm CNV
matrix.</p></li>
<li><p><code><a href="../reference/buildCNVTree.html">buildCNVTree()</a></code> will build the subclonality
tree.</p></li>
<li><p><code><a href="../reference/annotateCNVTree.html">annotateCNVTree()</a></code> will annotate each cnv gain/loss
in the tree.</p></li>
<li><p><code><a href="../reference/plotCNVTree.html">plotCNVTree()</a></code> will output the tree.</p></li>
</ul>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">tree_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CNVTree.html">CNVTree</a></span><span class="op">(</span><span class="va">scColon1</span>, values <span class="op">=</span> <span class="st">"calls"</span>, cnv_thresh <span class="op">=</span> <span class="fl">0.09</span>, healthyClusters <span class="op">=</span> <span class="st">"1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="fastCNV_sc_files/figure-html/tree_cnv-1.png" alt="" width="672"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Gadea Cabrejas, Clarice Groeneveld.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
