<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>CNVAnalysis Runs Copy Number Variation (CNV) analysis on a Seurat object or a list of Seurat objects. — CNVAnalysis • fastCNV</title><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CNVAnalysis Runs Copy Number Variation (CNV) analysis on a Seurat object or a list of Seurat objects. — CNVAnalysis"><meta name="description" content="This function performs CNV analysis by calculating genomic scores, applying optional denoising, and optionally
scaling the results based on a reference population. It processes single-cell or spatial transcriptomics data,
generating an additional assay with genomic scores and adding a new metadata column for CNV fractions."><meta property="og:description" content="This function performs CNV analysis by calculating genomic scores, applying optional denoising, and optionally
scaling the results based on a reference population. It processes single-cell or spatial transcriptomics data,
generating an additional assay with genomic scores and adding a new metadata column for CNV fractions."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">fastCNV</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/fastCNV_HD.html">Introduction to fastCNV - Visium HD data</a></li>
    <li><a class="dropdown-item" href="../articles/fastCNV_sc.html">Introduction to fastCNV - scRNAseq data</a></li>
    <li><a class="dropdown-item" href="../articles/fastCNV_ST.html">Introduction to fastCNV - ST data</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>CNVAnalysis Runs Copy Number Variation (CNV) analysis on a Seurat object or a list of Seurat objects.</h1>

      <div class="d-none name"><code>CNVAnalysis.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function performs CNV analysis by calculating genomic scores, applying optional denoising, and optionally
scaling the results based on a reference population. It processes single-cell or spatial transcriptomics data,
generating an additional assay with genomic scores and adding a new metadata column for CNV fractions.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">CNVAnalysis</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  referenceVar <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  referenceLabel <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pooledReference <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  scaleOnReferenceLabel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  assay <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  thresholdPercentile <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  geneMetadata <span class="op">=</span> <span class="fu"><a href="getGenes.html">getGenes</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  windowSize <span class="op">=</span> <span class="fl">150</span>,</span>
<span>  windowStep <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  saveGenomicWindows <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  topNGenes <span class="op">=</span> <span class="fl">7000</span>,</span>
<span>  chrArmsToForce <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genesToForce <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  regionToForce <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-object">object<a class="anchor" aria-label="anchor" href="#arg-object"></a></dt>
<dd><p>A Seurat object or a list of Seurat objects containing the data for CNV analysis. Each object
can be either <strong>single-cell</strong> or <strong>spatial transcriptomics</strong> data.</p></dd>


<dt id="arg-referencevar">referenceVar<a class="anchor" aria-label="anchor" href="#arg-referencevar"></a></dt>
<dd><p>The name of the metadata column in the Seurat object that contains reference annotations.</p></dd>


<dt id="arg-referencelabel">referenceLabel<a class="anchor" aria-label="anchor" href="#arg-referencelabel"></a></dt>
<dd><p>The label within <code>referenceVar</code> that specifies the reference population (can be any type of annotation).</p></dd>


<dt id="arg-pooledreference">pooledReference<a class="anchor" aria-label="anchor" href="#arg-pooledreference"></a></dt>
<dd><p>Logical. If <code>TRUE</code> (default), builds a pooled reference across all samples.</p></dd>


<dt id="arg-scaleonreferencelabel">scaleOnReferenceLabel<a class="anchor" aria-label="anchor" href="#arg-scaleonreferencelabel"></a></dt>
<dd><p>Logical. If <code>TRUE</code> (default), scales the results based on the reference population.</p></dd>


<dt id="arg-assay">assay<a class="anchor" aria-label="anchor" href="#arg-assay"></a></dt>
<dd><p>Name of the assay to run the CNV analysis on. Defaults to the results of <code>prepareCountsForCNVAnalysis</code> if available.</p></dd>


<dt id="arg-thresholdpercentile">thresholdPercentile<a class="anchor" aria-label="anchor" href="#arg-thresholdpercentile"></a></dt>
<dd><p>Numeric. Specifies the quantile range to consider (e.g., <code>0.01</code> keeps values between the 1st and 99th percentiles). Higher values filter out more background noise.</p></dd>


<dt id="arg-genemetadata">geneMetadata<a class="anchor" aria-label="anchor" href="#arg-genemetadata"></a></dt>
<dd><p>A dataframe containing gene metadata, typically from Ensembl.</p></dd>


<dt id="arg-windowsize">windowSize<a class="anchor" aria-label="anchor" href="#arg-windowsize"></a></dt>
<dd><p>Integer. Defines the size of genomic windows for CNV analysis.</p></dd>


<dt id="arg-windowstep">windowStep<a class="anchor" aria-label="anchor" href="#arg-windowstep"></a></dt>
<dd><p>Integer. Specifies the step size between genomic windows.</p></dd>


<dt id="arg-savegenomicwindows">saveGenomicWindows<a class="anchor" aria-label="anchor" href="#arg-savegenomicwindows"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, saves genomic window information in the current directory (default = <code>FALSE</code>).</p></dd>


<dt id="arg-topngenes">topNGenes<a class="anchor" aria-label="anchor" href="#arg-topngenes"></a></dt>
<dd><p>Integer. The number of top-expressed genes to retain in the analysis.</p></dd>


<dt id="arg-chrarmstoforce">chrArmsToForce<a class="anchor" aria-label="anchor" href="#arg-chrarmstoforce"></a></dt>
<dd><p>A chromosome arm (e.g., <code>"8p"</code>, <code>"3q"</code>) or a list of chromosome arms (e.g., <code>c("3q", "8p", "17p")</code>) to force into the analysis.
If specified, all genes within the given chromosome arm(s) will be included.</p></dd>


<dt id="arg-genestoforce">genesToForce<a class="anchor" aria-label="anchor" href="#arg-genestoforce"></a></dt>
<dd><p>A list of genes to force into the analysis (e.g. <code>c("FOXP3","MUC16","SAMD15")</code>).</p></dd>


<dt id="arg-regiontoforce">regionToForce<a class="anchor" aria-label="anchor" href="#arg-regiontoforce"></a></dt>
<dd><p>Chromosome region to force into the analysis (vector containing chr, start, end).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>If given a <strong>single</strong> Seurat object, returns the same object with:</p><ul><li><p>An <strong>additional assay</strong> containing genomic scores per genomic window.</p></li>
<li><p>A new <strong>CNV fraction column</strong> added to the object’s metadata.
If given a <strong>list</strong> of Seurat objects, returns the modified list.</p></li>
</ul></div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Gadea Cabrejas, Clarice Groeneveld.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

