<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>fastCNV calls all of the internal functions needed to compute the putative CNV on a Seurat object or a list of Seurat objects — fastCNV • fastCNV</title><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="fastCNV calls all of the internal functions needed to compute the putative CNV on a Seurat object or a list of Seurat objects — fastCNV"><meta name="description" content="This function orchestrates the CNV analysis on a Seurat object (or multiple objects). It calls internal functions such as
prepareCountsForCNVAnalysis, CNVAnalysis, CNVPerChromosomeArm, CNVcluster, and PlotCNVResults to compute the CNVs,
perform clustering, and generate heatmaps. The results are saved in the metadata of the Seurat object(s), with options for
generating and saving plots."><meta property="og:description" content="This function orchestrates the CNV analysis on a Seurat object (or multiple objects). It calls internal functions such as
prepareCountsForCNVAnalysis, CNVAnalysis, CNVPerChromosomeArm, CNVcluster, and PlotCNVResults to compute the CNVs,
perform clustering, and generate heatmaps. The results are saved in the metadata of the Seurat object(s), with options for
generating and saving plots."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">fastCNV</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/fastCNV_HD.html">Introduction to fastCNV - Visium HD data</a></li>
    <li><a class="dropdown-item" href="../articles/fastCNV_sc.html">Introduction to fastCNV - scRNAseq data</a></li>
    <li><a class="dropdown-item" href="../articles/fastCNV_ST.html">Introduction to fastCNV - ST data</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>fastCNV calls all of the internal functions needed to compute the putative CNV on a Seurat object or a list of Seurat objects</h1>

      <div class="d-none name"><code>fastCNV.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function orchestrates the CNV analysis on a Seurat object (or multiple objects). It calls internal functions such as
<code>prepareCountsForCNVAnalysis</code>, <code>CNVAnalysis</code>, <code>CNVPerChromosomeArm</code>, <code>CNVcluster</code>, and <code>PlotCNVResults</code> to compute the CNVs,
perform clustering, and generate heatmaps. The results are saved in the metadata of the Seurat object(s), with options for
generating and saving plots.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">fastCNV</span><span class="op">(</span></span>
<span>  <span class="va">seuratObj</span>,</span>
<span>  <span class="va">sampleName</span>,</span>
<span>  referenceVar <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  referenceLabel <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  assay <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  prepareCounts <span class="op">=</span> <span class="st">"default"</span>,</span>
<span>  aggregFactor <span class="op">=</span> <span class="fl">15000</span>,</span>
<span>  clusterResolution <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  aggregateByVar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  reCluster <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  pooledReference <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  scaleOnReferenceLabel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  thresholdPercentile <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  geneMetadata <span class="op">=</span> <span class="fu"><a href="getGenes.html">getGenes</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  windowSize <span class="op">=</span> <span class="fl">150</span>,</span>
<span>  windowStep <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  saveGenomicWindows <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  topNGenes <span class="op">=</span> <span class="fl">7000</span>,</span>
<span>  chrArmsToForce <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genesToForce <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  regionToForce <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  getCNVPerChromosomeArm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  getCNVClusters <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  k_clusters <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  h_clusters <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cellTypesToCluster <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mergeCNV <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mergeThreshold <span class="op">=</span> <span class="fl">0.98</span>,</span>
<span>  doPlot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  denoise <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  printPlot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  savePath <span class="op">=</span> <span class="st">"."</span>,</span>
<span>  outputType <span class="op">=</span> <span class="st">"png"</span>,</span>
<span>  clustersVar <span class="op">=</span> <span class="st">"cnv_clusters"</span>,</span>
<span>  splitPlotOnVar <span class="op">=</span> <span class="va">clustersVar</span>,</span>
<span>  referencePalette <span class="op">=</span> <span class="st">"default"</span>,</span>
<span>  clusters_palette <span class="op">=</span> <span class="st">"default"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seuratobj">seuratObj<a class="anchor" aria-label="anchor" href="#arg-seuratobj"></a></dt>
<dd><p>Seurat object or list of Seurat objects to perform the CNV analysis on.</p></dd>


<dt id="arg-samplename">sampleName<a class="anchor" aria-label="anchor" href="#arg-samplename"></a></dt>
<dd><p>Name of the sample or a list of names corresponding to the samples in the <code>seuratObj</code>.</p></dd>


<dt id="arg-referencevar">referenceVar<a class="anchor" aria-label="anchor" href="#arg-referencevar"></a></dt>
<dd><p>The variable name of the annotations in the Seurat metadata to be used as reference.</p></dd>


<dt id="arg-referencelabel">referenceLabel<a class="anchor" aria-label="anchor" href="#arg-referencelabel"></a></dt>
<dd><p>The label given to the observations you want as reference (can be any type of annotation).</p></dd>


<dt id="arg-assay">assay<a class="anchor" aria-label="anchor" href="#arg-assay"></a></dt>
<dd><p>Name of the assay to run the CNV on. Takes the results of <code>prepareCountsForCNVAnalysis</code> by default if available.</p></dd>


<dt id="arg-preparecounts">prepareCounts<a class="anchor" aria-label="anchor" href="#arg-preparecounts"></a></dt>
<dd><p>One of <code>TRUE</code>, <code>FALSE</code>, or <code>"default"</code> (default).
When <code>"default"</code>, <code>prepareCountsForCNVAnalysis</code> is run only if the number
of cells/spots is below 60,000.</p></dd>


<dt id="arg-aggregfactor">aggregFactor<a class="anchor" aria-label="anchor" href="#arg-aggregfactor"></a></dt>
<dd><p>The number of counts per spot desired (default = 15 000). If less than 1,000, will not run the <code>prepareCountsForCNVAnalysis</code> function.</p></dd>


<dt id="arg-clusterresolution">clusterResolution<a class="anchor" aria-label="anchor" href="#arg-clusterresolution"></a></dt>
<dd><p>The resolution wanted for the Seurat clusters (default = 0.8).</p></dd>


<dt id="arg-aggregatebyvar">aggregateByVar<a class="anchor" aria-label="anchor" href="#arg-aggregatebyvar"></a></dt>
<dd><p>If <code>referenceVar</code> is given, determines whether to use it to pool the observations (default = <code>TRUE</code>).</p></dd>


<dt id="arg-recluster">reCluster<a class="anchor" aria-label="anchor" href="#arg-recluster"></a></dt>
<dd><p>Whether to re-cluster if the Seurat object given already has a <code>seurat_clusters</code> slot in its metadata (default = <code>FALSE</code>).</p></dd>


<dt id="arg-pooledreference">pooledReference<a class="anchor" aria-label="anchor" href="#arg-pooledreference"></a></dt>
<dd><p>Default is <code>TRUE</code>. Will build a pooled reference across all samples if <code>TRUE</code>.</p></dd>


<dt id="arg-scaleonreferencelabel">scaleOnReferenceLabel<a class="anchor" aria-label="anchor" href="#arg-scaleonreferencelabel"></a></dt>
<dd><p>If <code>TRUE</code>, scales the results depending on the normal observations (default = <code>TRUE</code>).</p></dd>


<dt id="arg-thresholdpercentile">thresholdPercentile<a class="anchor" aria-label="anchor" href="#arg-thresholdpercentile"></a></dt>
<dd><p>Which quantiles to take (default 0.01). For example, <code>0.01</code> will take quantiles between 0.01-0.99. Background noise appears with higher numbers.</p></dd>


<dt id="arg-genemetadata">geneMetadata<a class="anchor" aria-label="anchor" href="#arg-genemetadata"></a></dt>
<dd><p>List of genes and their metadata (default uses genes from Ensembl version 113).</p></dd>


<dt id="arg-windowsize">windowSize<a class="anchor" aria-label="anchor" href="#arg-windowsize"></a></dt>
<dd><p>Size of the genomic windows for CNV analysis (default = 150).</p></dd>


<dt id="arg-windowstep">windowStep<a class="anchor" aria-label="anchor" href="#arg-windowstep"></a></dt>
<dd><p>Step between the genomic windows (default = 10).</p></dd>


<dt id="arg-savegenomicwindows">saveGenomicWindows<a class="anchor" aria-label="anchor" href="#arg-savegenomicwindows"></a></dt>
<dd><p>If <code>TRUE</code>, saves the information of the genomic windows in the current directory (default = <code>FALSE</code>).</p></dd>


<dt id="arg-topngenes">topNGenes<a class="anchor" aria-label="anchor" href="#arg-topngenes"></a></dt>
<dd><p>Number of top expressed genes to keep (default = 7000).</p></dd>


<dt id="arg-chrarmstoforce">chrArmsToForce<a class="anchor" aria-label="anchor" href="#arg-chrarmstoforce"></a></dt>
<dd><p>A chromosome arm (e.g., <code>"8p"</code>, <code>"3q"</code>) or a list of chromosome arms (e.g., <code>c("3q", "8p", "17p")</code>) to force into the analysis.</p></dd>


<dt id="arg-genestoforce">genesToForce<a class="anchor" aria-label="anchor" href="#arg-genestoforce"></a></dt>
<dd><p>A list of genes to force into the analysis (e.g. <code>c("FOXP3","MUC16","SAMD15")</code>).</p></dd>


<dt id="arg-regiontoforce">regionToForce<a class="anchor" aria-label="anchor" href="#arg-regiontoforce"></a></dt>
<dd><p>Chromosome region to force into the analysis (vector containing chr, start, end).</p></dd>


<dt id="arg-getcnvperchromosomearm">getCNVPerChromosomeArm<a class="anchor" aria-label="anchor" href="#arg-getcnvperchromosomearm"></a></dt>
<dd><p>If <code>TRUE</code>, will save the CNV per chromosome arm into the metadata.</p></dd>


<dt id="arg-getcnvclusters">getCNVClusters<a class="anchor" aria-label="anchor" href="#arg-getcnvclusters"></a></dt>
<dd><p>If <code>TRUE</code>, will perform clustering on the CNV scores and save them in the metadata of the Seurat object as <code>cnv_clusters</code>.</p></dd>


<dt id="arg-k-clusters">k_clusters<a class="anchor" aria-label="anchor" href="#arg-k-clusters"></a></dt>
<dd><p>Optional. Number of clusters to cut the dendrogram into. If <code>NULL</code>, the optimal number of clusters is determined automatically using the elbow method.</p></dd>


<dt id="arg-h-clusters">h_clusters<a class="anchor" aria-label="anchor" href="#arg-h-clusters"></a></dt>
<dd><p>Optional. The height at which to cut the dendrogram for clustering. If both <code>k</code> and <code>h</code> are provided, <code>k</code> takes precedence.</p></dd>


<dt id="arg-celltypestocluster">cellTypesToCluster<a class="anchor" aria-label="anchor" href="#arg-celltypestocluster"></a></dt>
<dd><p>Optional. Cell type annotations to use for clustering. If <code>NULL</code> (default), all observations will be clustered.</p></dd>


<dt id="arg-mergecnv">mergeCNV<a class="anchor" aria-label="anchor" href="#arg-mergecnv"></a></dt>
<dd><p>Logical. Whether to merge the highly correlated CNV clusters.</p></dd>


<dt id="arg-mergethreshold">mergeThreshold<a class="anchor" aria-label="anchor" href="#arg-mergethreshold"></a></dt>
<dd><p>A numeric value between 0 and 1. Clusters with correlation greater than this threshold will be merged. Default is 0.98.</p></dd>


<dt id="arg-doplot">doPlot<a class="anchor" aria-label="anchor" href="#arg-doplot"></a></dt>
<dd><p>If <code>TRUE</code>, will build a heatmap for each of the samples (default = <code>TRUE</code>).</p></dd>


<dt id="arg-denoise">denoise<a class="anchor" aria-label="anchor" href="#arg-denoise"></a></dt>
<dd><p>If <code>TRUE</code>, the denoised data will be used in the heatmap (default = <code>TRUE</code>).</p></dd>


<dt id="arg-printplot">printPlot<a class="anchor" aria-label="anchor" href="#arg-printplot"></a></dt>
<dd><p>If <code>TRUE</code>, the heatmap will be printed in the console (default = <code>FALSE</code>, the plot will only be saved in a PDF).</p></dd>


<dt id="arg-savepath">savePath<a class="anchor" aria-label="anchor" href="#arg-savepath"></a></dt>
<dd><p>Path to save the heatmap plot. If <code>NULL</code>, the plot won't be saved (default = <code>.</code>).</p></dd>


<dt id="arg-outputtype">outputType<a class="anchor" aria-label="anchor" href="#arg-outputtype"></a></dt>
<dd><p>Specifies the file format for saving the plot, either <code>"png"</code> or <code>"pdf"</code> (default = <code>"png"</code>).</p></dd>


<dt id="arg-clustersvar">clustersVar<a class="anchor" aria-label="anchor" href="#arg-clustersvar"></a></dt>
<dd><p>The variable name of the clusters in the Seurat metadata (default = <code>"cnv_clusters"</code>).</p></dd>


<dt id="arg-splitplotonvar">splitPlotOnVar<a class="anchor" aria-label="anchor" href="#arg-splitplotonvar"></a></dt>
<dd><p>The name of the metadata column to split the observations during the <code>plotCNVResults</code> step, if different from <code>referenceVar</code>.</p></dd>


<dt id="arg-referencepalette">referencePalette<a class="anchor" aria-label="anchor" href="#arg-referencepalette"></a></dt>
<dd><p>The color palette that should be used for <code>referenceVar</code> (default = <code>"default"</code>).</p></dd>


<dt id="arg-clusters-palette">clusters_palette<a class="anchor" aria-label="anchor" href="#arg-clusters-palette"></a></dt>
<dd><p>The color palette that should be used for <code>clustersVar</code> (default = <code>"default"</code>).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list of Seurat objects after all the analysis is complete. Heatmaps of the CNVs for every object in <code>seuratObj</code> are generated and saved in the specified path (default = current working directory).</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Gadea Cabrejas, Clarice Groeneveld.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

